packages/openapi-generator/src/file-serializer/schema.ts

```diff
@@ -0,0 +1,103 @@
+import { codeBlock } from '@sap-cloud-sdk/util';
+import {
+  OpenApiSchema,
+  OpenApiObjectSchema,
+  OpenApiObjectSchemaProperty
+} from '../openapi-types';
+import { getType } from '../parser/type-mapping';
+import {
+  isReferenceObject,
+  parseTypeNameFromRef,
+  isArraySchema,
+  isObjectSchema,
+  isEnumSchema,
+  isOneOfSchema,
+  isAllOfSchema,
+  isAnyOfSchema,
+  isNotSchema
+} from '../model';
+
+/**
+ * Serialize a schema.
+ * @param schema Parsed schema to be serialized.
+ * @returns The serialized schema
+ */
+export function serializeSchema(schema: OpenApiSchema): string {
+  if (isReferenceObject(schema)) {
+    return parseTypeNameFromRef(schema);
+  }
+  if (isArraySchema(schema)) {
+    const type = serializeSchema(schema.items);
+    return schema.uniqueItems ? `Set<${type}>` : `${type}[]`;
+  }
+  if (isObjectSchema(schema)) {
+    return serializeObjectSchema(schema);
+  }
+
+  if (isEnumSchema(schema)) {
+    return schema.enum.join(' | ');
+  }
+
+  if (isOneOfSchema(schema)) {
+    return schema.oneOf.map(type => serializeSchema(type)).join(' | ');
+  }
+
+  if (isAllOfSchema(schema)) {
+    return schema.allOf.map(type => serializeSchema(type)).join(' & ');
+  }
+
+  if (isAnyOfSchema(schema)) {
+    return schema.anyOf.map(type => serializeSchema(type)).join(' | ');
+  }
+
+  if (isNotSchema(schema)) {
+    return codeBlock`Except<any, ${serializeSchema(schema.not)}>`;
+  }
+
+  return getType(schema.type);
```

Comment:
Should we add a `isType` check here and throw an exception if no match was found to ensure we are not missing a schema type.

Reply:
We always serialize to any for unknown types.